lib/HiD/Role/IsConverted.pm

```diff
@@ -280,30 +312,19 @@ around BUILDARGS => sub {
   return $class->$orig( \%args );
 };
 
-{ # hide the map
 
-  ### FIXME make this extensible
-  my %conversion_extension_map = (
-    markdown => [ 'Text::Markdown'      , 'markdown' ] ,
-    mkdn     => [ 'Text::Markdown'      , 'markdown' ] ,
-    mk       => [ 'Text::Markdown'      , 'markdown' ] ,
-    md       => [ 'Text::Markdown'      , 'markdown' ] ,
-    mmd      => [ 'Text::MultiMarkdown' , 'markdown' ] ,
-    textile  => [ 'Text::Textile'       , 'process'  ] ,
-  );
+sub convert_by_extension {
+    my $self = shift;
 
-  sub _convert_by_extension {
-    my( $content , $extension ) = @_;
+    my $content = $self->content;
 
-    return $content
-      unless exists $conversion_extension_map{ $extension };
+    my $converter = $self->extension_processors->{ $self->ext }
+        or return $content;
 
-    my( $module , $method ) = @{ $conversion_extension_map{ $extension }};
+    my( $module , $method, @args ) = @$converter;
```

Comment:
`@args` doesn't appear to be used or documented?

Reply:
To be clear: not opposed to it being in there, but if it's an option, it should be documented in the attr description.

Reply:
Ah yes. This was my first stab at getting Mardent to work. I've added the `@args` to be able to pass static arguments for the method, à la Moose's "handles" currying. But at the end I didn't need it for my immediate goal. I'll remove it in 2 minutes.

Reply:
done.

Reply:
Ohyeah. If I ever need it, it'll come back. As it is, I try to YAGNI :-)